src/delete.ts

```diff
@@ -13,9 +13,17 @@ export function getVersionIds(input: Input): Observable<string[]> {
       input.owner,
       input.repo,
       input.packageName,
-      input.numOldVersionsToDelete,
+      input.numOldVersionsToDelete + input.numMostRecentVersionsToKeep,
       input.token
-    ).pipe(map(versionInfo => versionInfo.map(info => info.id)))
+    ).pipe(
+      map(versionInfo => {
+        const numberVersionsToDelete =
+          versionInfo.length - input.numMostRecentVersionsToKeep
+        return numberVersionsToDelete <= 0
+          ? []
+          : versionInfo.slice(0, numberVersionsToDelete).map(info => info.id)
```

Comment:
We could omit the supplementary ternary here and simply make this as follows; in the event of `[0, -n]` the `slice` should evaluate to an empty array.
 
```suggestion
        return versionInfo.slice(0, numberVersionsToDelete).map(info => info.id)
```

Reply:
As I understand,  end index being negative number indicates an offset from the end of the sequence. So, slice with negative end index will not result an empty array.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice

```
let a = [1,2,3,4,5];
console.log(a.slice(0,-2)); // print [1,2,3]
```

Reply:
Shoot! Missed `Math.max` call! I have updated the comment :yum:

Reply:
Also lets get remove that `numberVersionsToDelete` as well since we are referencing it only once with the above change